docs/parameters.md

```diff
@@ -0,0 +1,324 @@
+  planName: default
+  parameters:
+    - name: txt
+      value: Plain text, passed as a string
+    - name: meta
+      # Explicitly tells that the value is a valid JSON which should be 
+      # passed as a JSON object to the broker
+      type: json
+      valueFrom:
+        secretKeyRef:
+          name: mysecret
+          key: credentials
+```
+
+#### Other primitive types (int, bool, float)
+
+```yaml
+  ...
+  parameters:
+    - name: text
+      value: 1 # string by default, will be sent as "1"
+    - name: number
+      type: int
+      value: 1 # will be sent without quotes
+    - name: flag
+      type: bool
+      value: true
+    - name: fl
+      type: float
+      value: 123.5
+```
+
+#### Arrays
+A separate field `array: true` can be used to declare a value as an array. The
+element type is defined by the `type` value as described above.
+
+```yaml
+  ...
+  parameters:
+    - name: tags
+      array: true
+      value:
+      - foo
+      - bar
+    - name: jsonArray
+      type: json
+      array: true
+      value: [{"a": 1, "b": 2}]
+```
+
+### Passing all parameters from a single source at once
+
+In addition to support for explicitly specifying every parameter value fetched from 
+a particular source, Service Catalog also supports passing multiple parameters at 
+once using a `parametersFrom` field (similar to how `envFrom` works for `Pod`s).
+Instead of specifying a reference to a single `Secret` (or `ConfigMap`) key, you 
+just set the reference to the `Secret` (or `ConfigMap`), and all keys will be 
+automatically processed as separate parameters.
+
+In other words, instead of 
+```yaml
+  ...
+  parameters:
+    - name: key1
+      valueFrom:
+        secretKeyRef:
+          name: mysecret
+          key: key1
+    - name: key2
+      valueFrom:
+        secretKeyRef:
+          name: mysecret
+          key: key2
+     ...
+```
+you can just set it once in a `parametersFrom` field:
+```yaml
+  ...
+  parametersFrom:
+    - secretRef:
+        name: mysecret
+```
+
+You can also pass several `Secret`s and/or `ConfigMap`s at once:
+```yaml
+  ...
+  parametersFrom:
+    - secretRef:
+        name: secret1
+    - secretRef:
+        name: secret2
+    - configMapRef:
+        name: configmap1
+```
+
+Parameter values could be passed as JSON objects, the same way as for [individual parameters](#json-object):
+```yaml
+  ...
+  parametersFrom:
+    - type: json
+      secretRef:
+        name: mysecret
+```
+
+### Conflict resolution
+
+Within the scope of a `parameters` or a `parametersFrom` block, if there are 
+duplicate top-level names then the last one in each section will take precedence.
+If there are duplicate top-level names across `parameters` and `parametersFrom` 
+blocks then the `parameters` always takes precedence over `parametersFrom`.
+
+This conflict resolution strategy is consistent with the one supported by Kubernetes
+for environment variables (`env` and `envFrom` fields) in the `Pod` resource.
```

Comment:
IMO this amount of overriding is crossing the line into "overly complex". If people don't have an existing secret with the info they want then they should create one. Asking us to perform some manipulation to transform it seems like overkill to me because this implies the secret wasn't created for this purpose and the user is just reusing it. Which is fine, but then they should do the work of transforming it, not us.